pyserini/vectorizer.py

```diff
@@ -0,0 +1,64 @@
+import math
+import numpy as np
+from typing import List
+from scipy.sparse import csr_matrix
+from pyserini.search import pysearch
+from pyserini.index import pyutils
+
+
+class TfidfVectorizer:
+    def __init__(self, lucene_index_path: str, min_df: int = 0) -> None:
+        self.min_df: int = min_df
+        self.index_utils = pyutils.IndexReaderUtils(lucene_index_path)
+
+        # get num_docs
+        self.searcher = pysearch.SimpleSearcher(lucene_index_path)
+        self.num_docs: int = self.searcher.num_docs
+
+        # pre-processing
+        self.compute_idf_and_build_vocab()
+
+    def compute_idf_and_build_vocab(self):
```

Comment:
shouldn't this just be part of init?

Reply:
I thought that maybe we want to refactor this for later. 

merging with the constructor for now and will refactor later if there's the need.